msu/classes/weighted_container.nut

```diff
@@ -1,25 +1,27 @@
 ::MSU.Class.WeightedContainer <- class
 {
 	Total = null;
-	Array = null;
-	ApplyIdx = null;
+	Table = null;
 	Forced = null;
+	ItemsArray = null;
```

Comment:
Needs initialization

Reply:
> I don't think we should have _both_ the array and the table. If we wanna switch this over and there are things a table means we can't do, that's fine, just remove those functions. I don't wanna duplicate this stuff in memory unless we have to (like with ordered map)

The array is necessary for a foolproof foreach iteration (in order to allow a `foreach (item, weight in container)`. Otherwise I have to create an iteration array at the start of the foreach loop which reduces performance as I have to iterate over the entire table to first populate the array.

Ideally I wanted a `foreach (weight, item in container)` iteration but I haven't been able to do that. It might be doable if I create two arrays from the table at the start of the foreach. But that is really messy.

Alternatively we just disallow a foreach iteration?

Reply:
Yeah, that's okay for me. If we keep apply/map/filter functions those are sufficient anyway.